src/main/java/DNAnalyzer/protein/ProteinFinder.java

```diff
@@ -12,77 +12,95 @@
 package DNAnalyzer.protein;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
-import DNAnalyzer.codon.*;
-import DNAnalyzer.aminoAcid.*;
+
+import DNAnalyzer.aminoAcid.AminoAcidMapping;
+import DNAnalyzer.aminoAcid.AminoAcidNames;
+import DNAnalyzer.codon.CodonDataUtils;
 
 /**
  * Find proteins in a DNA sequence (contains the main algorithm).
  *
  * @author Piyush Acharya (@Verisimilitude11)
+ * @author Fernando Boaglio (@boaglio)
  * @version 1.2.1
  */
 
 public class ProteinFinder {
-    /**
-     * Amino acid list
-     */
 
-    private final List<String> aminoAcidList;
     /**
-     * protein list
+     * Utility classes should not have public constructors
      */
-    private final List<String> proteinList;
-
-    /**
-     * ProteinFinder default constructor to initialize aminoAcidList and proteinList
-     */
-    public ProteinFinder() {
-        aminoAcidList = new ArrayList<>();
-        proteinList = new ArrayList<>();
+    private ProteinFinder() {
     }
 
+    private static final int DNA_SIZE = 3;
```

Comment:
What does DNA_SIZE represent? The code looks great; I'll approve it after some testing later today!

Reply:
I think this variable name is not clear, it should reference nucleotides number of the DNA sequence. Maybe QT_NUCLEOTIDES ?

Reply:
Yes, that's a good variable name.